packages/api/src/platforms/vtex/clients/search/index.ts

```diff
@@ -156,6 +156,19 @@ export const IntelligentSearch = (
     }
   }
 
+  function getRequestInit(ctx: Context) {
```

Comment:
What about using the existing `withCookie({...})` method to get the header and pass it to the request? This way, we can keep the same pattern for both clients (commerce and client).

Also, I think the `forwardedHost` logic present there can be important to be in the header as well.